### PR TITLE
Add special holidays categories support

### PR DESCRIPTION
**Pull request by KJhellico**
## Proposed change

- Add special holidays categories support (special_public_holidays, special_bank_holidays etc)
- Add "workday" category, for official state holidays that are not non-working days (October 28 in Slovakia as an example)

## Type of change

- [ ] New country/market holidays support (thank you!)
- [ ] Supported country/market holidays update (calendar discrepancy fix, localization)
- [ ] Existing code/documentation/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency upgrade (version update)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [x] New feature (new python-holidays functionality in general)

## Checklist

- [x] I've followed the [contributing guidelines][contributing-guidelines]
- [x] This PR is filed against `beta` branch of the repository
- [x] This PR doesn't contain any merge conflicts and has clean commit history
- [x] The code style looks good: `make pre-commit`
- [x] All tests pass locally: `make test`, `make tox` (we strongly encourage adding tests to your code)
- [x] The related [documentation][docs] has been added/updated (check off the box for free if no updates is required)

[contributing-guidelines]: https://github.com/dr-prodigy/python-holidays/blob/beta/CONTRIBUTING.rst
[docs]: https://github.com/dr-prodigy/python-holidays/tree/beta/docs/source

### CODE REVIEW

**Comment by PPsyrius** [comment]
> `workday` category

That would really come in handy for all those "officially observed but not a non-working day list" for Thailand's "National Day of Importance" 20+ entries here:
image

Also, can you clarify a bit more on `special_public_holidays` - these are functionally the same as `special_holidays`, just that these only apply to the `PUBLIC` category, right?

File: holidays/constants.py [hunk 1]
```diff
@@ -42,4 +43,5 @@
     ISLAMIC,
     PUBLIC,
     SCHOOL,
+    WORKDAY,
```
Comment (arkid15r) [inline holidays/constants.py:46]: The category list keeps growing. It has to be documented when/how each of the categories should be used. Ideally with some use case examples.

**Review by arkid15r — COMMENTED** [review]
LGTM

File: holidays/constants.py (hunk 1 above)
Comment (arkid15r) [reply holidays/constants.py:46]: I'll work w/ you on that soon, filed #1348 for now.

**Pull request merged**